src/js/components/AppListComponent.jsx

```diff
@@ -178,7 +178,7 @@ var AppListComponent = React.createClass({
     var filterLabels = filters[FilterTypes.LABELS];
     var filterStatus = filters[FilterTypes.STATUS];
 
-    if (!Util.isEmptyString(filterText)) {
+    if (filterText != null && filterText !== "") {
```

Comment:
So, ```filterText``` is ```undefined``` if the user hasn't typed in before.
```javascript
  isEmptyString: function (str) {
    return this.isString(str) && (str == null || str === "");
  },
```
```this.isString(undefined)``` returns ```false```.
I negated it with ```!Util.isEmptyString(undefined)```.
Alles klar? :)